var/spack/repos/builtin/packages/qhull/package.py

```diff
@@ -47,8 +48,7 @@ class Qhull(Package):
     
     depends_on('cmake')
 
-    def install(self, spec, prefix):
-        with working_dir('spack-build', create=True):
-            cmake('..', *std_cmake_args)
-            make()
-            make("install")
+    @CMakePackage.sanity_check('build')
+    @CMakePackage.on_package_attributes(run_tests=True)
```

Comment:
> A lot of packages refuse to build in the root directory. A lot also refuse to build in a different directory. I'm not a big fan of CMake...

i would say that's because they have bad CMakeList.txt and hard-coded something. If everything is right, you should certainly be able to build out-of-sources with no issues. 

If there is a package which uses CMake in a weird way and does not allow to build out-of-source, then one could re-implement `def wdir(self)` to fit this.